File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        //edge case
        if(nums.size() == 1) return -1;

        //sort the array
        sort(nums.begin(), nums.end());

        //check for duplicacy
        for(int i=1; i<nums.size(); i++)
        {
            if(nums[i-1] == nums[i])
            {
                return nums[i];
            }
        }
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int size = nums.length;
        int start = 0;

        while (start < size) {
            int correctIndex = nums[start] - 1;
            if (nums[start] != nums[correctIndex]) {
                int temp = nums[correctIndex];
                nums[correctIndex] = nums[start];
                nums[start] = temp;
            }
            else {
                start++;
            }
        }

        for (int i=0; i<size; i++) {
            if (i + 1 != nums[i]) {
                return nums[i];
            }
        }
        return -1;
    }
}